orangecontrib/spectroscopy/widgets/gui.py

```diff
@@ -474,3 +477,152 @@ def set_default(self, v):
     def focusInEvent(self, *e):
         self.focusIn.emit()
         return QWidget.focusInEvent(self, *e)
+
+
+class VerticalPeakLine(pg.InfiniteLine):
+
+    """pyqtgraph.InfiniteLine with adjustments for spectra analysis
+    """
+
+    def __init__(self, pos=None, angle=90, pen=None, movable=True,
+                 bounds=None, label=None, span=None):
+        super().__init__(pos, angle, pen, movable, bounds)
+        self.moving = False
+        self.mouseHovering = False
+        self.span = span
+        self._boundingRect = None
+        self._endPoints = [0, 1]
+        self._bounds = None
+        self._lastViewSize = None
+
+        if label is None:
+            self.label = VerticalPeakLineLabel(self,
+                                               text=str(self.getXPos()), position=(1))
+
+    def setSpan(self, mn, mx):
+        if self.span != (mn, mx):
+            self.span = (mn, mx)
+            self.update()
+
+    def _computeBoundingRect(self):
```

Comment:
Also, this function is mostly copied, what is the difference? Is the difference needed at all? If needed, but predictable, maybe the `super()` function can be called and then the output could be adapted.